### ISSUE TITLE
OCommandExecutionException when expanding empty distinct

### ISSUE THREAD

**Issue by johansjob** [orientechnologies/orientdb#5374]
I get a OCommandExecutionException when doing:
```
Select expand(distinct(@rid)) from []
```
Or
```
Select expand(distinct(@rid)) from (select from vertex where name = "this name doesn't exist")
```

It works fine with only the expand or only the distinct, just not combined.

I'm guessing the problem is more generic than just the distinct function..

**Comment by welsayedaly** [comment]
the expand() function expects a collection or a link field. And with the distinct() function you will retrieve unique data that does not point to any data.

**Comment by johansjob** [comment]
Yes precisely, still shouldn't throw an exception in my opinion. There are situation where you don't know if the subquery returns any results or not. In these cases you expect an empty expand and not an exception. 

A workaround, like this, is an option but shouldn't be needed.
´´´
Select expand(dist) from (select from distinct(@rid) as dist from (select from vertex where name = "this name doesn't exist"))
´´´

**Comment by johansjob** [comment]
Can confirm it is more generic than the distinct function.

If you try to expand a function that returns null you get an exception. Adding a separate variable in another select, as shown above, is a workaround confirming it really is a bug.

**Comment by luigidellaquila** [comment]
Hi @johansjob 

This issue is already fixed in both 2.1.17 and v.2.2 (latest stable versions)

I'm closing this ticket, please reopen if you still see similar problems

Thanks

Luigi
